### PR TITLE
feat: add dip to spiritnet runtime

### PR DESCRIPTION
**Pull request by Ad96el**
## fixes [#3250](https://github.com/KILTprotocol/ticket/issues/3250)

Once the benchmark PR is merged, I would update the weights.

### CODE REVIEW

File: runtimes/peregrine/src/lib.rs [hunk 1]
```diff
@@ -851,6 +850,8 @@ impl InstanceFilter<RuntimeCall> for ProxyType {
 							| did::Call::submit_did_call { .. }
 							| did::Call::update_deposit { .. }
 							| did::Call::change_deposit_owner { .. }
+							| did::Call::create_from_account { .. }
+							| did::Call::dispatch_as { .. }
```
Comment (Ad96el) [inline runtimes/peregrine/src/lib.rs:854]: Seems like we forgot about these two new extrinsics ._.

Comment (ntn-x2) [reply runtimes/peregrine/src/lib.rs:854]: Yeah, I think we need to somehow factor them out in a way that we update it once, and changes are reflected in other places. We tend to forget the proxy stuff quite often.

File: runtimes/spiritnet/Cargo.toml [hunk 2]
```diff
@@ -179,6 +185,8 @@ std = [
   "pallet-authorship/std",
   "pallet-balances/std",
   "pallet-collective/std",
+  "pallet-dip-provider/std",
+  "pallet-deposit-storage/std",
```
Comment (ntn-x2) [inline runtimes/spiritnet/Cargo.toml:189]: Can you put them in the right order? It makes it easier to see if we miss some features.

Comment (ntn-x2) [reply runtimes/spiritnet/Cargo.toml:189]: Like, where's the `std` for `pallet-configuration` and `kilt-runtime-api-dip-provider`?

File: runtimes/spiritnet/src/lib.rs [hunk 3]
```diff
@@ -817,6 +821,7 @@ impl InstanceFilter<RuntimeCall> for ProxyType {
 							| delegation::Call::change_deposit_owner { .. }
 					)
 					| RuntimeCall::Democracy(..)
+					// Excludes `DepositStorage`
```
Comment (ntn-x2) [inline runtimes/spiritnet/src/lib.rs:824]: I didn't see this for Peregrine? Can we add it?

File: runtimes/spiritnet/src/dip/mod.rs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	type Identifier = DidIdentifier;
+	// The identity commitment is defined as the Merkle root of the linked identity
+	// info, as specified by the [`LinkedDidInfoProvider`].
+	type IdentityCommitmentGenerator = DidMerkleRootGenerator<Runtime>;
+	// Identity info is defined as the collection of DID keys, linked accounts, and
+	// the optional web3name of a given DID subject.
+	type IdentityProvider = LinkedDidInfoProvider<MAX_LINKED_ACCOUNTS>;
+	type ProviderHooks = DepositCollectorHooks;
+	type RuntimeEvent = RuntimeEvent;
+	// TODO: update if benchmark pr is merged
```
Comment (ntn-x2) [inline runtimes/spiritnet/src/dip/mod.rs:50]: REMINDER

File: runtimes/spiritnet/src/dip/mod.rs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	type CheckOrigin = EnsureSigned<AccountId>;
+	// The balances pallet is used to reserve/unreserve tokens.
+	type Currency = Balances;
+	type DepositHooks = DepositHooks;
+	type MaxKeyLength = ConstU32<MAX_DEPOSIT_PALLET_KEY_LENGTH>;
+	type Namespace = DepositNamespace;
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeHoldReason = RuntimeHoldReason;
+	// TODO: update if benchmark pr is merged
+	type WeightInfo = ();
```
Comment (ntn-x2) [inline runtimes/spiritnet/src/dip/mod.rs:68]: REMINDER

**Review by ntn-x2 — COMMENTED** [review]
Looks good so far!

File: runtimes/spiritnet/src/lib.rs (hunk 3 above)
Comment (Ad96el) [reply runtimes/spiritnet/src/lib.rs:824]: If you mean the comment, it is already present: https://github.com/KILTprotocol/kilt-node/pull/616/files#diff-56f0782a52c09d5ac94e1b327f8afaae9dd45f76ed0c17e75773c74ed433caf8R836. 

Or what do you mean?

File: runtimes/spiritnet/src/dip/mod.rs (hunk 5 above)
Comment (Ad96el) [reply runtimes/spiritnet/src/dip/mod.rs]: [1d13410](https://github.com/KILTprotocol/kilt-node/pull/616/commits/1d13410c4dc25a7305fc6ef082be7b86b4f16458)

File: runtimes/spiritnet/src/dip/mod.rs (hunk 4 above)
Comment (Ad96el) [reply runtimes/spiritnet/src/dip/mod.rs]: [1d13410](https://github.com/KILTprotocol/kilt-node/pull/616/commits/1d13410c4dc25a7305fc6ef082be7b86b4f16458)

File: runtimes/spiritnet/Cargo.toml (hunk 2 above)
Comment (Ad96el) [reply runtimes/spiritnet/Cargo.toml]: Changed order in: [8810616](https://github.com/KILTprotocol/kilt-node/pull/616/commits/8810616671f2aa36e1f36563e1443e05516fb510)

The only differences between the spiritnet cargo.toml vs. the peregrine cargo.toml should come from the sudo and configuration pallet.

File: runtimes/spiritnet/Cargo.toml [hunk 6]
```diff
@@ -36,8 +36,12 @@ attestation.workspace = true
 ctype.workspace = true
 delegation.workspace = true
 did.workspace = true
+kilt-runtime-api-dip-provider.workspace = true
 kilt-support.workspace = true
+pallet-configuration.workspace = true
```
[the remark this answers is not in the document]
Comment (Ad96el) [reply runtimes/spiritnet/Cargo.toml]: removed in: [3f4af68](https://github.com/KILTprotocol/kilt-node/pull/616/commits/3f4af683d73dfff1a0884eb9914025e5bfa6196d)

**Review by ntn-x2 — APPROVED** [review]

**Pull request merged**